lib/Futures/SharedState.h

```diff
@@ -163,18 +163,14 @@ class SharedState {
             return;
           }
           TRI_ASSERT(state == State::OnlyResult);  // race with setResult
-#ifndef _MSC_VER
           [[fallthrough]];
-#endif
 
         case State::OnlyResult:
-          if (_state.compare_exchange_strong(state, State::Done, std::memory_order_acquire)) {
+          if (_state.compare_exchange_strong(state, State::Done, std::memory_order_release)) {
```

Comment:
Actually, the `memory_order_acquire` here was correct. We only reach this line if the previous CAS in line 161 has failed. But that means that the value we have loaded in line 157 (with `memory_order_acquire`) has been overwritten by some other thread in `setResult`. So we have to use `memory_order_acquire` here to synchronize with the release-CAS in `setResult` (line 198). (Alternatively, we could change the CAS in line 161 to use `memory_order_acquire` for the reload in the failure case to establish the synchronize-with relation at that point. Then we could use `memory_order_relaxed` for this CAS).

Reply:
Thanks.
Fixed via https://github.com/arangodb/arangodb/pull/10291/commits/31044a399c259be5780b467a19dc25dd2b0e7d2f